### PR TITLE
[Method References 2] LPS-165626

### PR DESCRIPTION
**Pull request by JanisZhang**
Previous pr:https://github.com/liferay-core-infra/liferay-portal/pull/2434

### CODE REVIEW

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/DefaultJournalTransformerListenerRegistryImpl.java [hunk 1]
```diff
@@ -38,39 +38,31 @@
 
 	@Override
 	public List<TransformerListener> getTransformerListeners() {
-		return ListUtil.filter(
-			new ArrayList<>(_transformerListeners.values()),
-			transformerListener -> {
-				if (transformerListener.isEnabled()) {
-					return true;
-				}
+		List<TransformerListener> transformerListeners =
```
Comment (dantewang) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/transformer/DefaultJournalTransformerListenerRegistryImpl.java:41]: @JanisZhang This method can be simplified with

```
return ListUtil.filter(
	_serviceTrackerList.toList(), TransformerListener::isEnabled);
```

please update, thanks.

File: modules/apps/journal/journal-service/src/main/java/com/liferay/journal/model/impl/JournalArticleImpl.java [hunk 2]
```diff
@@ -98,9 +100,23 @@ public static String getContentByLocale(
...
+		List<TransformerListener> transformerListeners =
+			JournalTransformerListenerRegistryUtil.getTransformerListeners();
+
+		Stream<TransformerListener> transformerListenersStream =
+			transformerListeners.stream();
+
 		TransformerListener transformerListener =
-			JournalTransformerListenerRegistryUtil.getTransformerListener(
-				LocaleTransformerListener.class.getName());
+			transformerListenersStream.filter(
```
Comment (dantewang) [inline modules/apps/journal/journal-service/src/main/java/com/liferay/journal/model/impl/JournalArticleImpl.java:110]: @JanisZhang Could you change it to an ordinary for loop?

**Comment by JanisZhang** [comment]
ci:stop

**Comment by dantewang** [comment]
ci:test:sf

**Review by dantewang — APPROVED** [review]

**Comment by JanisZhang** [comment]
Hi @zhangyan0701, could you please double-check the integration test failure? The test you mentioned can be passed in my local. Thanks! cc @dantewang

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@JanisZhang no regressions found.

**Comment by JanisZhang** [comment]
thanks @catarineaguiar !

**Review by dantewang — APPROVED** [review]

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/transformer/test/JournalTransformerTest.java [hunk 3]
```diff
@@ -319,10 +320,26 @@ protected Map<String, String> getTokens() throws Exception {
 	}
 
 	protected void initRegexTransformerListener() {
-		TransformerListener transformerListener =
-			_journalTransformerListenerRegistry.getTransformerListener(
-				"com.liferay.journal.internal.transformer." +
-					"RegexTransformerListener");
+		List<TransformerListener> transformerListeners =
```
Comment (tinatian) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/transformer/test/JournalTransformerTest.java:323]: @JanisZhang Please mimic the solution at https://github.com/liferay-core-infra/liferay-portal/blob/8567ab84e3ba352f600659394a4dee861a74cdaf/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java#L6978

**Comment by tinatian** [comment]
@dantewang @JanisZhang Please see my comment and resend.

Thanks.
Tina

**Pull request closed without merging**